### PR TITLE
Typos correction

### PR DESCRIPTION
**Pull request by irinakhismatullina**
Data preparation scripts and baseline model code added

### CODE REVIEW

**Comment by smola** [comment]
@irinakhismatullina Do not include `.DS_Store` files on your commits.
You can also create a global gitignore to avoid the issue in the future: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: Typos correction/IPython Notebooks/typos_functions.ipynb [hunk 1]
```diff
@@ -0,0 +1,207 @@
+{
```
Comment (smola) [inline Typos correction/IPython Notebooks/typos_functions.ipynb]: We usually avoid spaces and uppercase in directory and file names. so `typo-correction` and `ipython-notebooks` would look better (or with underscore, I'm not sure if there is any convention for this, maybe underscore is preferred in Python world, cc @vmarkovtsev).

File: Typos correction/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+Baseline model ready. Short description:
```
Comment (smola) [inline Typos correction/README.md]: Use markdown formatting in readmes (e.g. sections with `##`, `###`, etc).

File: Typos correction/baseline.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+from tqdm import tqdm
+from sklearn.ensemble import RandomForestClassifier
+from multiprocessing import Pool, cpu_count
+
+import pysymspell.symspell as symspell
+from typos_functions import *
+
+MAX_DISTANCE = 2
+
+class Baseline():
```
Comment (smola) [inline Typos correction/baseline.py]: I think `class Baseline:` (no parens) in Python 3 is preferred.

File: Typos correction/create_typos.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import pandas
+import numpy
+import itertools
+import random
+from tqdm import tqdm
+from typos_functions import rand_bool
+
+letters = list('qwertyuiopasdfghjklzxcvbnm')
```
Comment (EgorBu) [inline Typos correction/create_typos.py]:
```python
import string
letters =  list(string.ascii_lowercase)
```
right now it's not obvios that you are not missing anything.

File: Typos correction/data_preparation.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import pandas
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: the same comment about order of imports as above.

File: Typos correction/data_preparation.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import numpy
+import itertools
+import sys
+import modelforge
+
+path_to_sourced = '/Users/irina/Documents/GitHub/ml'
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: delete hardcode.

File: Typos correction/data_preparation.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+sys.path.append(path_to_sourced)
+import sourced
+from sourced.ml.models import id2vec
+
+def flattify(dataframe, column, new_column, apply_function = lambda x:x):
+    """
+    Flattify dataframe on 'column' with extracted elemenents put to 'new_column'
+    """
+    other_columns = list(dataframe.columns)
+    flat_other = numpy.repeat(dataframe.loc[:, other_columns].values, repeats = numpy.array(dataframe[column].apply(lambda x:len(apply_function(x))).tolist()), axis=0)
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: use `pep8 --max-line-length=99 .` to check style - this line is too long.

File: Typos correction/data_preparation.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    id_stats = id_info.loc[:, ['num_files', 'num_occ', 'num_repos', 'identifier']]
+    id_stats = id_stats.groupby(['identifier']).sum()
+    id_stats.to_csv(stats_file)
+    return id_stats
+  
+def embedding(model, token):
+    return model.embeddings[model['i.' + token]]
+
+if __name__ == "__main__":
+    data = pandas.read_csv('repos2ids_v3.4_stats.csv')
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: hardcode - should be a parameter.

File: Typos correction/data_preparation.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    id_info = get_info(data, 'id_info.csv')  
+    id_stats = get_stats(id_info, 'id_stats.csv')
+
+    model = id2vec.Id2Vec()
+    model.load('emb-18.asdf')
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: the same.

File: Typos correction/data_preparation.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+    common_tokens = list(sorted(list(set(id_stats.index.tolist()).intersection(set([x[2:] for x in model.tokens])))))
+    vector_matrix = numpy.array([embedding(model, token) for token in common_tokens])
+
+    vector_matrix.dump('vector_matrix')
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: double quotes instead of single quotes.

File: Typos correction/data_preparation.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    common_tokens.dump('common_tokens')
```
Comment (EgorBu) [inline Typos correction/data_preparation.py]: new line in the end.

File: Typos correction/filter_identifiers.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+def check_split(split, tokens_set): 
+    if type(split) != str:
+        return False
+
+    for token in split.split():
+        if token not in tokens_set:
+            return False
+    return True
+
```
Comment (EgorBu) [inline Typos correction/filter_identifiers.py]: two lines between functions.

File: Typos correction/get_frequencies.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import numpy
+import pandas
+
```
Comment (EgorBu) [inline Typos correction/get_frequencies.py]: two lines

File: Typos correction/main.py [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+import os
+import sys
+
+from create_typos import create_typos
+from baseline import baseline
+from filter_identifiers import filter_identifiers
+from get_frequencies import get_frequencies
+from pick_subset import pick_subset
+
+class ArgumentDefaultsHelpFormatterNoNone(argparse.ArgumentDefaultsHelpFormatter):
```
Comment (EgorBu) [inline Typos correction/main.py]: is it copy-paste from src-d/ml? You can use import.

File: Typos correction/pick_subset.py [hunk 15]
```diff
@@ -0,0 +1,8 @@
+import pandas
+from typos_functions import rand_bool
```
Comment (EgorBu) [inline Typos correction/pick_subset.py]: imports.

File: Typos correction/pick_subset.py [hunk 16]
```diff
@@ -0,0 +1,8 @@
...
+
+def pick_subset(args):
+    data = pandas.read_pickle(args.input_file)
+    pick_indices = [rand_bool(args.picked_portion) for i in range(len(data))]
+    data = data[pick_indices]
+    data.to_pickle(args.out_file)
```
Comment (EgorBu) [inline Typos correction/pick_subset.py]: new line

File: Typos correction/README.md (hunk 2 above)
Comment (bzz) [reply Typos correction/README.md]: I find these resources helpful, when need to refresh something about MD
 - https://guides.github.com/features/mastering-markdown/
 - https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

Comment (irinakhismatullina) [reply Typos correction/README.md]: Thanks, very helpful

**Comment by vmarkovtsev** [comment]
@irinakhismatullina can you please place everything under `lookout/style/typos/research`

**Pull request merged**